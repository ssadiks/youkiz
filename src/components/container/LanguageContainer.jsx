import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import { setLocaleAction } from '../../redux/actions';
import { LOCALES } from '../../constants';

class LanguageContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      locale: props.locale
    };
  }

  onChangeLocale = (event, index, value) => {
    this.setState({ locale: value });
    this.props.setLocaleAction(value);
  }

  render() {
    return (
      <SelectField
        value={this.state.locale}
        onChange={this.onChangeLocale}
        floatingLabelText="Country"
        className="country"
      >
        {
          LOCALES.map(locale => (
            <MenuItem
              key={locale}
              value={locale}
              primaryText={locale}
            />
          ))
        }

      </SelectField>
    );
  }
}

LanguageContainer.defaultProps = {
};

LanguageContainer.PropTypes = {
  setLocaleAction: PropTypes.func.isRequired,
  locale: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  locale: state.translationReducer.locale
});

export default connect(mapStateToProps, { setLocaleAction })(LanguageContainer);
