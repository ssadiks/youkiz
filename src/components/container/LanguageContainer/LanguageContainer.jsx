import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setLocaleAction } from '../../../redux/actions/index';
import Language from '../../presentational/Language/Language';

class LanguageContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      locale: props.locale
    };
  }

  onChangeLocale = (event, index, value) => {
    this.setState({ locale: value });
    this.props.setLocaleAction(value);
  }

  render() {
    return <Language locale={this.state.locale} onChangeLocale={this.onChangeLocale} />;
  }
}

LanguageContainer.PropTypes = {
  setLocaleAction: PropTypes.func.isRequired,
  locale: PropTypes.string.isRequired
};

const mapStateToProps = state => ({
  locale: state.translationReducer.locale
});

export default connect(mapStateToProps, { setLocaleAction })(LanguageContainer);
