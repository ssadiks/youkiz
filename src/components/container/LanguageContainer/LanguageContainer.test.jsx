import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Adapter from 'enzyme-adapter-react-16';
import LanguageContainer from './LanguageContainer';

configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const INITIAL_STATE = {
  translationReducer: {
    locale: 'es'
  }
};

describe('LanguageContainer', () => {
  it('renders with default props', () => {
    const store = mockStore(INITIAL_STATE);
    const wrapper = shallow(
      <LanguageContainer
        store={store}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
