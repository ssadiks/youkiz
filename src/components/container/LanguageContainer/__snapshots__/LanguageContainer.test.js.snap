// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageContainer renders with default props 1`] = `
<LanguageContainer
  locale="es"
  setLocaleAction={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
